### DISCUSSION TITLE
SSH client certs with ed25519

### DISCUSSION THREAD

**Discussion by logopk** [smallstep/certificates#569] [Q&A] [answered]
Hi,

all step ssh certificates seem to be ecdsa.

The host key can be created using crypto --kty=OKP --curve=ed25519,  but the step ssh certificate does not have these options.

Should this be possible, or is this missing? If so, are you planning to get the more secure ed25519 anytime soon?

Thanks

Peter

**Comment by maraino** [comment, marked as answer]
This should be possible, as a workaround you can create a certificate from a previously created key using the `--sign` flag:

```sh
# create private (ed) and public (ed.pub) keys
ssh-keygen -t ed25519 -f ed
# Sign ed.pub
step ssh certificate --sign mariano@smallstep.com ed.pub
```
Or with step only:
```sh
# create keypair
step crypto keypair --kty OKP ed25519.pem ed25519.priv
# Convert public key to ssh format
step crypto key format --ssh --out ed25519.pub ed25519.pem
# Sign ed25519.pub
step ssh certificate --sign mariano@smallstep.com ed25519.pub
```

You can do the same for hosts using for example the previously generated key in `/etc/sshd` and using `--sign --host`

**Reply by logopk** [reply]
Perfect. Thanks, that helps already.

**Comment by maraino** [comment]
I've added this new issue to track the feature in the cli https://github.com/smallstep/cli/issues/477
